Model/NodeType/Category.php

```diff
@@ -97,16 +106,26 @@ public function fetchData(array $nodes, $storeId)
 
         $localNodes = [];
         $categoryIds = [];
+        $categories = [];
 
         foreach ($nodes as $node) {
             $localNodes[$node->getId()] = $node;
-            $categoryIds[] = (int)$node->getContent();
+            $categoryId = (int)$node->getContent();
+            $categoryIds[] = $categoryId;
+            $categories[$categoryId] = $this->getCategory($categoryId);
         }
 
         $categoryUrls = $this->getResource()->fetchData($storeId, $categoryIds);
 
         $this->profiler->stop(__METHOD__);
 
-        return [$localNodes, $categoryUrls];
+        return [$localNodes, $categoryUrls, $categories];
+    }
+
+    public function getCategory($categoryId)
+    {
+        $category = $this->categoryFactory->create();
+        $category->load($categoryId);
```

Comment:
As I said before - use a category collection to load all needed categories from DB at once.